src/charm.py

```diff
@@ -33,83 +32,59 @@ def __init__(self, *args):
             args: Arguments passed to the CharmBase parent constructor.
         """
         super().__init__(*args)
-        self.framework.observe(self.on.httpbin_pebble_ready, self._on_httpbin_pebble_ready)
+        try:
+            self._charm_state = CharmState.from_charm(charm=self)
+        except CharmConfigInvalidError as exc:
+            self.model.unit.status = ops.BlockedStatus(exc.msg)
+            return
```

Comment:
I'm wondering if `CharmState` is relevant here as it's a copy of `SmtpRelationData` validation, so doesn't add any value

Reply:
I expect it to diverge when the `smtp` relation is added, so perhaps we can defer the decision until then

Reply:
Fine for me